src/librustdoc/html/render/mod.rs

```diff
@@ -1869,69 +1871,100 @@ pub(crate) fn render_impl_summary(
+            Primitive(it) => it.render_into(f),
+            Union(it) => it.render_into(f),
+            Enum(it) => it.render_into(f),
+            Typedef(it) => it.render_into(f),
+            Module(it) => it.render_into(f),
+            ForeignType(it) => it.render_into(f),
+            _ => Ok(()),
+        };
+        res.map_err(|e| {
+            let askama::Error::Fmt(e) = e else {
+                unreachable!("yaml, json and ? are not in use")
+            };
+            e
+        })
     }
+}
 
+fn print_sidebar(cx: &Context<'_>, it: &clean::Item, buffer: &mut Buffer) {
+    let inner = match *it.kind {
+        clean::StructItem(ref s) => SidebarKind::Struct(sidebar_struct(cx, it, s)),
+        clean::TraitItem(ref t) => SidebarKind::Trait(sidebar_trait(cx, it, t)),
+        clean::PrimitiveItem(_) => SidebarKind::Primitive(sidebar_primitive(cx, it)),
+        clean::UnionItem(ref u) => SidebarKind::Union(sidebar_union(cx, it, u)),
+        clean::EnumItem(ref e) => SidebarKind::Enum(sidebar_enum(cx, it, e)),
+        clean::TypedefItem(_) => SidebarKind::Typedef(sidebar_typedef(cx, it)),
+        clean::ModuleItem(ref m) => SidebarKind::Module(sidebar_module(&m.items)),
+        clean::ForeignTypeItem => SidebarKind::ForeignType(sidebar_foreign_type(cx, it)),
+        _ => SidebarKind::Other,
+    };
     // The sidebar is designed to display sibling functions, modules and
     // other miscellaneous information. since there are lots of sibling
     // items (and that causes quadratic growth in large modules),
     // we refactor common parts into a shared JavaScript file per module.
     // still, we don't move everything into JS because we want to preserve
     // as much HTML as possible in order to allow non-JS-enabled browsers
     // to navigate the documentation (though slightly inefficiently).
-
-    if !it.is_mod() {
-        let path: String = cx.current.iter().map(|s| s.as_str()).intersperse("::").collect();
-
-        write!(buffer, "<h2><a href=\"index.html\">In {}</a></h2>", path);
-    }
-
-    // Closes sidebar-elems div.
-    buffer.write_str("</div>");
+    let (title_prefix, title) = if it.is_struct()
+        || it.is_trait()
+        || it.is_primitive()
+        || it.is_union()
+        || it.is_enum()
+        || it.is_mod()
+        || it.is_typedef()
+    {
+        (
+            match *it.kind {
+                clean::ModuleItem(..) if it.is_crate() => "Crate ",
+                clean::ModuleItem(..) => "Module ",
+                _ => "",
+            },
+            it.name.as_ref().unwrap().as_str(),
+        )
+    } else {
+        ("", "")
+    };
+    let empty = String::new();
+    let version =
+        if it.is_crate() { cx.cache().crate_version.as_ref() } else { None }.unwrap_or(&empty);
+    let path = if !it.is_mod() {
+        cx.current.iter().map(|s| s.as_str()).intersperse("::").collect::<String>()
+    } else {
+        String::new()
+    };
```

Comment:
```suggestion
    let path: String = if !it.is_mod() {
        cx.current.iter().map(|s| s.as_str()).intersperse("::").collect()
    } else {
        "".into()
    };
```

Reasoning: the existing code here used a type annotation of `: String` instead of the turbofish. Keeping the type annotation allows us to avoid the turbofish, avoid repeating the `String` type in the other branch, and makes it more immediately clear what the type of `path` is.